src/safeposix/syscalls/fs_calls.rs

```diff
@@ -389,41 +389,75 @@ impl Cage {
+    /// * `path` - This argument points to a pathname naming the file.
+    //             For example: "/parentdir/file" represents the new file name as "file", which will be created at this path (/parentdir/file).
+    /// * `mode` - The mode argument specifies both the permissions to use and the type of node to be created. It is a combination (using bitwise OR) of one of the file types and the permissions for the new node.
+    ///            FileType - In LIND, we have only implemented the file type of "Character Device" represented by S_IFCHR flag.
+    ///            FilePermission - The general permission mode used is "S_IRWXA": which represents the read, write, and search permissions on the new file.
+    ///            The final file mode is represented by the bitwise-OR of FileType and FilePermission Flags.
+    /// * `dev` -  It is a configuration-dependent specification of a character or block I/O device. If mode does not indicate a block special or character special device, dev is ignored.
+    ///            Since "CharDev" is the only supported type, 'dev' is represented using makedev() function; that returns a formatted device number   
+    ///            For example: "makedev(&DevNo { major: majorId, minor: minorId })" accepts a Device Number that consists of: 
+    ///            MajorID, identifying the class of the device, and a minor ID, identifying a specific instance of a device in that class.
+    ///
+    /// ### Returns
+    /// Upon successful creation of the file, 0 is returned.
+    /// Otherwise, errors or panics are returned for different scenarios.
+    ///
+    /// ### Errors
+    /// * `ENOENT` - occurs when a directory component in the absolute path does not exist
+    /// * `EPERM` - the mode bits for the new file are not sane
+    /// * `EINVAL` - when any other file type (regular, socket, block, fifo) instead of character file type is passed
+    /// * `EEXIST` - when the file to be created already exists
+    /// 
+    /// for more detailed description of all the commands and return values, see 
+    /// [mknod(2)](https://man7.org/linux/man-pages/man2/mknod.2.html)
+    ///
     pub fn mknod_syscall(&self, path: &str, mode: u32, dev: u64) -> i32 {
-        //Check that path is not empty
+        // Check that the given input path is empty
         if path.len() == 0 {
             return syscall_error(Errno::ENOENT, "mknod", "given path was null");
         }
+        // Retrieve the absolute path from the root directory. The absolute path is then used to validate directory paths
+        // while navigating through subdirectories and establishing new directory at the given location.
         let truepath = normpath(convpath(path), self);
 
-        //pass the metadata to this helper. If passed table is none, then create new instance
+        // Store the FileMetadata into a helper variable which is used for fetching the metadata of a given inode from the Inode Table. 
```

Comment:
wrap this and other comments to be <80 chars wide

Reply:
Done